src/main/java/org/opensearch/agent/tools/CreateAlertTool.java

```diff
@@ -0,0 +1,282 @@
+            StringSubstitutor substitute = new StringSubstitutor(tmpParams, "${parameters.", "}");
+            String finalToolPrompt = substitute.replace(TOOL_PROMPT_TEMPLATE);
+            tmpParams.put("prompt", finalToolPrompt);
+
+            RemoteInferenceInputDataSet inputDataSet = RemoteInferenceInputDataSet.builder().parameters(tmpParams).build();
+            ActionRequest request = new MLPredictionTaskRequest(
+                modelId,
+                MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inputDataSet).build()
+            );
+            client.execute(MLPredictionTaskAction.INSTANCE, request, ActionListener.wrap(r -> {
+                ModelTensorOutput modelTensorOutput = (ModelTensorOutput) r.getOutput();
+                Map<String, ?> dataMap = Optional
+                    .ofNullable(modelTensorOutput.getMlModelOutputs())
+                    .flatMap(outputs -> outputs.stream().findFirst())
+                    .flatMap(modelTensors -> modelTensors.getMlModelTensors().stream().findFirst())
+                    .map(ModelTensor::getDataAsMap)
+                    .orElse(null);
+                if (dataMap == null) {
+                    throw new IllegalArgumentException("No dataMap returned from LLM.");
+                }
+                String response = "";
+                if (dataMap.containsKey("response")) {
+                    response = (String) dataMap.get("response");
+                    Pattern jsonPattern = Pattern.compile("```json(.*?)```", Pattern.DOTALL);
+                    Matcher jsonBlockMatcher = jsonPattern.matcher(response);
+                    if (jsonBlockMatcher.find()) {
+                        response = jsonBlockMatcher.group(1);
+                        response = response.replace("\\\"", "\"");
+                    }
+                } else {
+                    // LLM sometimes returns the tensor results as a json object directly instead of
+                    // string response, and the json object is stored as a map.
+                    response = StringUtils.toJson(dataMap);
+                }
+                if (!isJson(response)) {
+                    throw new IllegalArgumentException(
+                        LoggerMessageFormat.format(null, "The response from LLM is not a json: [{}]", response)
+                    );
+                }
+                listener.onResponse((T) response);
+            }, e -> {
+                log.error("Failed to run model " + modelId, e);
+                listener.onFailure(e);
+            }));
+        } catch (Exception e) {
+            log.error("Failed to call CreateAlertTool", e);
+            listener.onFailure(e);
+        }
+    }
+
+    private String getIndexMappingInfo(Map<String, String> parameters) throws InterruptedException, ExecutionException {
+        if (!parameters.containsKey("indices") || parameters.get("indices").isEmpty()) {
```

Comment:
When the parameters contains the key but value is null, `parameters.get("indices").isEmpty()` can throw NPE.